src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -710,26 +702,24 @@ private static void removeAllUnused(long threshold) throws IOException {
     }
 
     public void verifyConnection() throws IOException {
-      synchronized (this) {
```

Comment:
I think we still want this lock.  In an edge case, you could have a large number of calls all made to `checkApiUrlValidity()` at the same time and end up costing significant rate limit budget.

Reply:
ah sure